var/spack/repos/builtin/packages/eccodes/package.py

```diff
@@ -69,6 +71,12 @@ class Eccodes(CMakePackage):
         description="Specify JPEG2000 decoding/encoding backend",
     )
     variant("png", default=False, description="Enable PNG support for decoding/encoding")
+    variant(
+        "aec",
+        default=True,
+        when="@2.25.0:",
+        description="Enable Adaptive Entropy Coding for decoding/encoding",
+    )
     variant(
         "aec", default=False, description="Enable Adaptive Entropy Coding for decoding/encoding"
     )
```

Comment:
There's no issue with having two.  It looks like she wants to change the default in version `2.25.0:`. (WarpX also does this -- look at their `dims` variant). 

The only issue is that this currently does not work.  The variant can have only one default.  So you could change it to either:

```python
    variant(
        "aec",
        default=True,
        when="@2.25:",    # I would just skip the .0 here
        description="Enable Adaptive Entropy Coding for decoding/encoding",
    )
    variant(
        "aec",
        default=True,
        when="@:2.24",
        description="Enable Adaptive Entropy Coding for decoding/encoding",
    )
```

Or have one variant directive with your preferred default.  The issue to track for the multiple default problem is #38302; I'm working on fixing that but am not quite done.

Reply:
Ah.  I didn't realize that was the goal here.

Reply:
Thanks for the reviews and for the background info. Chosen option 1 as recommended, with order flipped such that default would be `True`. https://github.com/spack/spack/pull/40770/commits/ff66d579364056ddd870c86232a7c49c98a7c145

Reply:
I think we should default `aec` to `True` unconditionally and have a `conflict` for old versions that don't have the feature.

Reply:
Hi @skosukhin I think a conflict is not required because aec was a feature since atleast 2.0.0 https://github.com/ecmwf/eccodes/blob/2.0.0/CMakeLists.txt#L66-L70 which covers all the versions spack supports for eccodes.